compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -851,9 +851,7 @@ impl<'ast, 'ra, 'tcx> LateResolutionVisitor<'_, 'ast, 'ra, 'tcx> {
         }
 
         // Try to find in last block rib
-        if let Some(rib) = &self.last_block_rib
-            && let RibKind::Normal = rib.kind
-        {
+        if let Some(rib) = &self.last_block_rib {
```

Comment:
Will this modification cause the code to display contextual tips like: "binding `a` is accessible in a different scope within the same function"?

```rs
fn main() {
    
    {
        let a = 1;
        struct S;
    }
    dbg!(a);
}

```